crates/db_schema/src/impls/images.rs

```diff
@@ -0,0 +1,78 @@
+use crate::{
+  newtypes::{DbUrl, LocalUserId},
+  schema::{
+    local_image::dsl::{local_image, local_user_id, pictrs_alias},
+    remote_image::dsl::{link, remote_image},
+  },
+  source::images::{LocalImage, LocalImageForm, RemoteImage, RemoteImageForm},
+  utils::{get_conn, DbPool},
+};
+use diesel::{
+  dsl::exists,
+  insert_into,
+  result::Error,
+  select,
+  ExpressionMethods,
+  NotFound,
+  QueryDsl,
+  Table,
+};
+use diesel_async::RunQueryDsl;
+use url::Url;
+
+impl LocalImage {
+  pub async fn create(pool: &mut DbPool<'_>, form: &LocalImageForm) -> Result<Self, Error> {
+    let conn = &mut get_conn(pool).await?;
+    insert_into(local_image)
+      .values(form)
+      .get_result::<Self>(conn)
+      .await
+  }
+
+  pub async fn get_all_by_local_user_id(
+    pool: &mut DbPool<'_>,
+    user_id: &LocalUserId,
+  ) -> Result<Vec<Self>, Error> {
+    let conn = &mut get_conn(pool).await?;
+    local_image
+      .filter(local_user_id.eq(user_id))
+      .select(local_image::all_columns())
+      .load::<LocalImage>(conn)
+      .await
+  }
+
+  pub async fn delete_by_alias(pool: &mut DbPool<'_>, alias: &str) -> Result<usize, Error> {
+    let conn = &mut get_conn(pool).await?;
+    diesel::delete(local_image.filter(pictrs_alias.eq(alias)))
+      .execute(conn)
+      .await
+  }
+}
+
+impl RemoteImage {
+  pub async fn create(pool: &mut DbPool<'_>, links: Vec<Url>) -> Result<usize, Error> {
```

Comment:
I'm not too sure of what this one's for. These are for images *not stored* in our pictrs? 

How does it work with the `cache_external_image`, and `image_proxy` settings? 

Is it considered remote if its a proxied image? If that's the case then you need to make sure this `create` isn't run if proxy_images = false.

Reply:
https://github.com/LemmyNet/lemmy/pull/4035#issuecomment-1876980133